pkg/mover/chart.go

```diff
@@ -256,12 +257,68 @@ func (cm *ChartMover) loadChartFromIntermediateBundle(bundlePath string) error {
 // loadChartFromPath load the chart in memory from a given path
 func (cm *ChartMover) loadChartFromPath(path string) error {
 	var err error
-	if cm.chart, err = loader.Load(path); err != nil {
-		return &ChartLoadingError{Path: path, Inner: err}
+	dir := path
+	if isChartFile(path) {
+		// always unpack the chart tarball to a directory to avoid issues loading
+		// files if the tgz contains replicated entries
+		// https://github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/issues/111
+		if dir, err = expandedChartDir(path); err != nil {
+			return err
+		}
+		defer os.RemoveAll(filepath.Dir(dir)) // clean it up on function exit
+	}
+
+	if cm.chart, err = loader.Load(dir); err != nil {
+		return &ChartLoadingError{Path: dir, Inner: err}
 	}
 	return nil
 }
 
+func isChartFile(path string) bool {
+	info, err := os.Stat(path)
+	return err == nil && !info.IsDir()
+}
+
+func expandedChartDir(path string) (string, error) {
+	tmpdir, err := os.MkdirTemp("", "expanded-chart-*")
+	if err != nil {
+		return "", fmt.Errorf("failed to make temporary directory: %w", err)
+	}
+
+	if err := chartutil.ExpandFile(tmpdir, path); err != nil {
+		return "", fmt.Errorf("failed to extract chart: %w", err)
+	}
+
+	chartname, err := detectChartName(tmpdir)
+	if err != nil {
+		return "", fmt.Errorf("failed to detect chartname from extracted chart: %w", err)
+	}
+	return filepath.Join(tmpdir, chartname), nil
+}
+
+// detectChartName expects chartdir to contain a single subdirectory, matching
+// the chartname
```

Comment:
I'd elaborate a little bit more, explaining that a justs extracted Helm chart usually contains a directory in the root called with the name of the chart.

Reply:
I can link to the helm chart docs, in fact I thought I had done it...

Reply:
done